test/test_roscpp/test/test_transport_tcp.cpp

```diff
@@ -132,7 +134,7 @@ TEST_F(Synchronous, readWhileWriting)
   for (int i = 0; i < 10; ++i)
   {
     const uint32_t buf_size = 1024*1024;
-    std::auto_ptr<uint8_t> read_buf(new uint8_t[buf_size]);
+    boost::shared_ptr<uint8_t[]> read_buf(new uint8_t[buf_size]);
```

Comment:
I know that `ros_comm` generally uses boost, but given that this is buried in a test, can we just use `std::shared_ptr`?

Reply:
Since the package isn't building with C++11 it can't use the `std` smart pointers. And changing the C++ standard version was beyond the scope of this test.